src/hsd/dict.py

```diff
@@ -13,24 +13,49 @@
     QUOTING_CHARS, SPECIAL_CHARS
 from hsd.eventhandler import HsdEventHandler, HsdEventPrinter
 
-_ItemType = Union[float, int, bool, str]
+_ItemType = Union[float, complex, int, bool, str]
 
 _DataType = Union[_ItemType, List[_ItemType]]
 
 # Pattern to transform HSD string values into actual Python data types
 _TOKEN_PATTERN = re.compile(r"""
 # Integer
-(?:\s*(?:^|(?<=\s))(?P<int>[+-]?[0-9]+)(?:\s*$|\s+))
+(?:
+    \s*(?:^|(?<=\s))
+    (?P<int>[+-]?[0-9]+)
+    (?:\s*$|\s+)
+)
 |
 # Floating point
-(?:\s*(?:^|(?<=\s))
-(?P<float>[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?)(?:$|(?=\s+)))
+(?:
+    \s*(?:^|(?<=\s))
+    (?P<float>[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?)
+    (?:$|(?=\s+))
+)
+|
+# Complex value
+(?:
+    \s*(?:^|(?<=\s))
+    (?P<complex>
+        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?i
+        |
+        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[-+][0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?i
```

Comment:
```suggestion
    (?P<complex>
        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[ij]
        |
        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[-+][0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[ij]
```

Reply:
Well, I just saw that you specifically mentioned in dftb+ that "j" is invalid, so I guess you can ignore the comment then. ;)